source/common/filesystem/inotify/watcher_impl.cc

```diff
@@ -87,10 +87,10 @@ void WatcherImpl::onInotifyEvent() {
         if (watch.events_ & events) {
           if (watch.file_ == file) {
             ENVOY_LOG(debug, "matched callback: file: {}", file);
-            watch.cb_(events);
+            THROW_IF_NOT_OK(watch.cb_(events));
```

Comment:
Heh, it's mildly amusing that the fist change in this "removing exceptions" PR is a change with *adds* a throw where there was none before :P